recipes/qt/6.x.x/conanfile.py

```diff
@@ -522,26 +526,133 @@ def package_id(self):
+            self.cpp_info.components[pluginname].includedirs = []
+            if "Core" not in requires:
+                requires.append("Core")
+            self.cpp_info.components[pluginname].requires = requires
+
+        core_reqs = ["zlib::zlib"]
+        if self.options.with_pcre2:
+            core_reqs.append("pcre2::pcre2")
+        if self.options.with_doubleconversion:
+            core_reqs.append("double-conversion::double-conversion")
+        if self.options.get_safe("with_icu", False):
+            core_reqs.append("icu::icu")
+
+        _create_module("Core", core_reqs)
+        self.cpp_info.components["Core"].libs.append("Qt6Core_qobject%s" % libsuffix)
+        if self.options.gui:
+            gui_reqs = []
+            if self.options.with_freetype:
+                gui_reqs.append("freetype::freetype")
+            if self.options.with_libpng:
+                gui_reqs.append("libpng::libpng")
+            if self.options.get_safe("with_fontconfig", False):
+                gui_reqs.append("fontconfig::fontconfig")
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                gui_reqs.append("xorg::xorg")
+                if not tools.cross_building(self, skip_x64_x86=True):
+                    gui_reqs.append("xkbcommon::xkbcommon")
+            if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") != "no":
+                gui_reqs.append("opengl::opengl")
+            _create_module("Gui", gui_reqs)
+        if self.options.with_sqlite3:
+            _create_plugin("QSQLiteDriverPlugin", "qsqlite", "sqldrivers", ["sqlite3::sqlite3"])
+        if self.options.with_pq:
+            _create_plugin("QPSQLDriverPlugin", "qsqlpsql", "sqldrivers", ["libpq::libpq"])
+        if self.options.with_odbc:
+            if self.settings.os != "Windows":
+                _create_plugin("QODBCDriverPlugin", "qsqlodbc", "sqldrivers", ["odbc::odbc"])
+        _create_module("Network", ["openssl::openssl"] if self.options.openssl else [])
+        _create_module("PrintSupport", ["Gui", "Widgets"])
+        _create_module("Sql")
+        _create_module("Test")
+        if self.options.widgets:
+            _create_module("Widgets", ["Gui"])
+        if self.options.get_safe("opengl", "no") != "no":
+            _create_module("OpenGL", ["Gui"])
+        if self.options.widgets and self.options.get_safe("opengl", "no") != "no":
+            _create_module("OpenGLWidgets", ["OpenGL", "Widgets"])
+        _create_module("DBus")
+        _create_module("Concurrent")
+        _create_module("Xml")
+
+        if self.options.qt5compat:
+            _create_module("Core5Compat")
+
+        if self.options.qtdeclarative:
+            _create_module("Qml", ["Network"])
+            _create_module("QmlModels", ["Qml"])
+            _create_module("Quick", ["Gui", "Qml", "QmlModels"])
+            _create_module("QuickWidgets", ["Gui", "Qml", "Quick", "Widgets"])
```

Comment:
There's a few of these Using widgets but not checking `self.options.widgets` is this a problem?

Reply:
You're right, I won't create components which require a component which does not exist